### DISCUSSION TITLE
Is it safe to `import * as chains`?

### DISCUSSION THREAD

**Discussion by frolic** [wagmi-dev/references#197] [Q&A] [answered]
I want to use `@wagmi/chains` as a peer dependency in my library, so that users can update at will to pick up new chains. As such, I'd like a safe/consistent way to import all chains in my library code, but I am hesitant about `import * as chains` in case `@wagmi/chains` ends up exporting more than just the list of chains (it already does this for the `Chain` type but that gets compiled out).

Is there a recommended way to get a list or object with all chains?

**Comment by tmm** [comment, marked as answer]
`@wagmi/chains` is only used for exporting Chains (and any required types, e.g. `Chain`) so other runtime code will not get pulled in when you do a namespace import. One thing you might want to check is tree-shakeability across popular bundlers for namespace imports.

**Reply by frolic** [reply]
Ahh, I hadn't considered the tree-shakeability of this kind of import. Is that why ya'll removed `defaultChains` etc. from wagmi?

**Reply by tmm** [reply]
that's why we removed the `chain` object (e.g. `chain.mainnet`, `chain.optimism`)
